op-service/eth/types.go

```diff
@@ -325,15 +334,19 @@ func BlockAsPayload(bl *types.Block, shanghaiTime *uint64) (*ExecutionPayload, e
 		BlobGasUsed:   (*Uint64Quantity)(bl.BlobGasUsed()),
 	}
 
-	if shanghaiTime != nil && uint64(payload.Timestamp) >= *shanghaiTime {
+	if config.ShanghaiTime != nil && uint64(payload.Timestamp) >= *config.ShanghaiTime {
 		payload.Withdrawals = &types.Withdrawals{}
 	}
 
+	if config.IsthmusTime != nil && uint64(payload.Timestamp) >= *config.IsthmusTime {
+		payload.WithdrawalsRoot = bl.Header().WithdrawalsHash
+	}
+
 	return payload, nil
 }
 
-func BlockAsPayloadEnv(bl *types.Block, shanghaiTime *uint64) (*ExecutionPayloadEnvelope, error) {
-	payload, err := BlockAsPayload(bl, shanghaiTime)
+func BlockAsPayloadEnv(bl *types.Block, config *params.ChainConfig) (*ExecutionPayloadEnvelope, error) {
```

Comment:
We now need to check for IsthmusTime as well before setting WithdrawalsRoot in the payload, so we pass in ChainConfig.